### ISSUE TITLE
XSLT 2 Help

### ISSUE THREAD

**Issue by JustinCampbell9** [ebeshero/DHClass-Hub#563]
When adding the elements to the list, I cannot get them to be each independent elements. I'm not quite sure what I'm doing wrong.

**Comment by ghbondar** [comment]
Can you post your template code? @JustinCampbell9

**Comment by JustinCampbell9** [comment]
<html>
            <head>
                <title>Digital Mitford</title>
            </head>
            <body>
                <ol>
                    <li>
                        <h1>Archives Holding Mitford's Papers</h1>
                        <ul>
                            <xsl:apply-templates select=".//div/listOrg/org/orgName"/>
                        </ul>
                    </li>
                </ol>
            </body>
        </html>



I was trying to format it like the website but I'm not sure if that's messing anything up

**Comment by ghbondar** [comment]
@JustinCampbell9  Here's a couple things:
1) The contents of your <h1>-element is from the xml-document's <listOrg>-element, so you should be grabbing that with a template.
2) Remember that each set of lists needs to cycle or loop-through its own template, so plan on having multiple templates, each matching-on and outputting one kind of data. So, you have an outer list of headings (your <h1>-elements from the <listOrg>-elements), each containing a different inner list.
3) Then you have other templates retrieving the data for the inner lists.

**Comment by ghbondar** [comment]
@JustinCampbell9 So, the XPath inside your select-attribute is too long and complicated: you only need it to grad the text of those `<head>`-elements inside the `listOrg`-elements.

**Comment by ghbondar** [comment]
@JustinCampbell9 Please view my replies inside GitHub, since the formatting got screwed-up.

**Comment by JustinCampbell9** [comment]
So essentially I shouldn't be containing the headers and such in lists

**Comment by ghbondar** [comment]
@JustinCampbell9  Check out [my posted example from Friday](https://github.com/ebeshero/DHClass-Hub/blob/master/Class-Examples/XSLT/xslt-day-2-example.xsl) for an example on using multiple templates:  

[https://github.com/ebeshero/DHClass-Hub/blob/master/Class-Examples/XSLT/xslt-day-2-example.xsl](https://github.com/ebeshero/DHClass-Hub/blob/master/Class-Examples/XSLT/xslt-day-2-example.xsl)

**Comment by JustinCampbell9** [comment]
I really don't understand the Xpath that I'm using. There's only a single head element in the listOrg. This only gives me "<ol>Archives Holding Mitford’s PapersOrganizations Relevant to Mitford’s WorldFictional Groups Referenced by Mitford</ol>" as an output

**Comment by ebeshero** [comment]
@JustinCampbell9 You're right that there is only one `<head>` element in each `<listOrg>`. That's by design and it's the head labelling each list. This is a little confusing the first time you do it, and part of it is working out how to code an HTML list. Those go like this:

For the entire list, you need a wrapper element, either `<ol>` (for an ordered, numbered list) or `<ul>` (for an unordered, bulleted list). You want just ONE ordered list, which will output those head elements as its items. When you want just ONE of something, you set up that one element in the template match on the document node: (This always works because there's one and only one document node!)

```
<xsl:template match="/">
       <ol>   
               <xsl:apply-templates select="descendant::listOrg"> 
<!-- This makes ONE outer list -->
      </ol>

</xsl:template>
```

Okay, you now need to write a template to match on the `<listOrg>` elements (wherever they are). These will fire three times, and every time they fire, you want those to be wrapped in HTML `<li>` elements. You also want to start your new, inner list in that template matching on `<listOrg>`. Try it and see how the output looks.

**Comment by ebeshero** [comment]
@JustinCampbell9 just updated my comment, so you'll want to view it on GitHub. Hope this helps.
